testkit-backend/backend.go

```diff
@@ -94,6 +97,38 @@ func (g GenericTokenManager) HandleSecurityException(_ context.Context, token ne
 	return handled, nil
 }
 
+type TestKitClientCertificateProvider struct {
+	id      string
+	backend *backend
+}
+
+func NewTestKitClientCertificateProvider(id string, backend *backend) (*TestKitClientCertificateProvider, error) {
```

Comment:
:upside_down_face: `NewTestKitClientCertificateProvider`'s signature suggests is fallible (whereas it de-facto isn't). On top, the only call site I could find ignores the error return type. I suggest to change the signature.

Reply:
Absolutely that's a copy and paste from NewStaticClientCertificateProvider 👍